test/e2e/util/requestserving.go

```diff
@@ -34,22 +34,31 @@ func SetupRequestServingNodePools(ctx context.Context, t *testing.T, kubeconfigP
 			mgmtNodePools = append(mgmtNodePools, nodePool)
 		}
 	}
-	g.Expect(len(mgmtNodePools) >= 2).To(BeTrue(), "we need at least 2 AWS management cluster nodepools in different zones")
+	g.Expect(len(mgmtNodePools) >= 3).To(BeTrue(), "we need at least 3 AWS management cluster nodepools in different zones")
 
-	var nodePoolA, nodePoolB *hyperv1.NodePool
+	var nodePoolA, nodePoolB, nodePoolC *hyperv1.NodePool
 	nodePoolA = &mgmtNodePools[0]
 	for i := range mgmtNodePools[1:] {
 		nodePool := &mgmtNodePools[1+i]
 		if nodePool.Spec.Platform.AWS.Subnet.ID != nodePoolA.Spec.Platform.AWS.Subnet.ID {
-			nodePoolB = nodePool
-			break
+			if nodePoolB == nil {
+				nodePoolB = nodePool
+			}
+			if nodePoolB != nil && nodePool.Spec.Platform.AWS.Subnet.ID != nodePoolB.Spec.Platform.AWS.Subnet.ID {
+				nodePoolC = nodePool
+				break
+			}
 		}
 	}
 	g.Expect(nodePoolB).ToNot(BeNil(), "did not find 2 nodepools with different subnets in parent")
+	g.Expect(nodePoolC).ToNot(BeNil(), "did not find 3 nodepools with different subnets in parent")
 
 	// Prepare and create nodepools for request serving components
 	reqServingNodePoolA := nodePoolA.DeepCopy()
 	reqServingNodePoolB := nodePoolB.DeepCopy()
+	nonReqServingNodePoolA := nodePoolA.DeepCopy()
```

Comment:
why do we need three of these?

Reply:
it's for etcd, each etcd pod needs to be in a different zone